### PR TITLE
lightweight model: create dummy settings file to follow. Fixes JB#28832

### PR DESCRIPTION
**Pull request by pvuorela**
QFileSystemWatcher requires paths and dirs to exist in order to follow
changes. Added error checking and made a dummy settings file to be
created if none existed to allow following.

### CODE REVIEW

File: lightweight/calendareventsmodel/calendareventsmodel.cpp [hunk 1]
```diff
@@ -64,9 +64,26 @@ NemoCalendarEventsModel::NemoCalendarEventsModel(QObject *parent) :
...
+        }
+    } else {
+        qWarning() << "CalendarEventsModel not following database changes, dir not found:" << privilegedDataDir;
+    }
+
     QSettings settings("nemo", "nemo-qml-plugin-calendar");
-    mWatcher->addPath(settings.fileName());
+
+    if (!QFile::exists(settings.fileName())) {
+        // forcefully create a settings file so changes can be followed
```
Comment (thp) [inline lightweight/calendareventsmodel/calendareventsmodel.cpp]: Would it be better to just touch the file instead of writing dummy values to it?

Comment (pvuorela) [inline lightweight/calendareventsmodel/calendareventsmodel.cpp]: Maybe, wasn't sure myself. Now kept actual file access behind one interface. Can change to touching.

**Comment by pvuorela** [comment]
Now creating file without QSettings.

**Comment by thp** [comment]
LGTM. Of course, we could also monitor the directory containing the file for changes and then subscribe to file changes once the file appears, but I guess this works as well.

**Pull request merged**